envi/archs/arm/disasm.py

```diff
@@ -192,12 +199,20 @@ def p_dp_imm_shift(opval, va):
     Rm = opval & 0xf
     shtype = (opval >> 5) & 0x3
     shval = (opval >> 7) & 0x1f   # effectively, rot*2
-
+    mnem = dp_mnem[ocode]
     if ocode in dp_noRn:# FIXME: FUGLY (and slow...)
-        olist = (
-            ArmRegOper(Rd, va=va),
-            ArmRegShiftImmOper(Rm, shtype, shval, va),
-        )
+        #is it a mov? Only if shval is a 0, type is lsl, and ocode = 13
+        if  (ocode == 13) and ((shval != 0) or (shtype != 0)):
+            mnem = dp_shift_mnem[shtype]
+            olist = (
+                ArmRegOper(Rd, va=va),
+                ArmRegShiftImmOperMov(Rm, shtype, shval, va),
```

Comment:
why is this not just three operands?
( ArmRegOper, ArmRegOper, ArmImmOper )
???  
the emulator gets to decide how to use them.